### ISSUE TITLE
Selecting a control plane/node version or constraint with a channel

### ISSUE THREAD

**Issue by jawnsy** [terraform-google-modules/terraform-google-kubernetes-engine#1356]
### TL;DR

https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/issues/1279 added the capability to select a specific version when channels are not in use (the special `UNSPECIFIED` designator).

I would like the ability to either describe a version constraint, or to select the newest version of the given channel (instead of just the default version of a given channel), as we can do in the UI and as indicated in this comment: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/issues/1279#issuecomment-1167584257

### Additional information

The UI selection for the Regular release channel looks like this:

![image](https://user-images.githubusercontent.com/52710/183316588-c799f5cd-010f-4690-8793-a97b72de7577.png)

When creating a cluster using this submodule, the default version will be selected, and GKE will track this version according to the maintenance policy. We can also manually upgrade our cluster.

However, there is no way to indicate that we would like to upgrade to the newest supported version of a given release channel, rather than simply using the default.

**Comment by bharathkkb** [comment]
Thanks for the FR @jawnsy. Happy to review a PR too if you have bandwidth.
